lib/pdksync/utils.rb

```diff
@@ -907,5 +914,73 @@ def self.analyse_jenkins_report(url, module_name, build_id)
       write_to_file("results_#{module_name}.out", @data)
       return_data
     end
+
+    # @summary
+    #   Check the most recent tagged release on GitHub for the gem
+    # @param [String] gem_to_test
+    #   The gem to test
+    #   The current version of the gem
+    def self.check_gem_latest_version(gem_to_test)
+      remote_version = Octokit.tags("puppetlabs/#{gem_to_test}").first[:name]
+    rescue StandardError => error
+      puts "(WARNING) Unable to check latest gem version. #{error}".red
+      return remote_version
+    end
+
+    # @summary
+    #   Update the gem version by one
+    # @param [String] gem_version
+    #   The current version of the gem
+    #   The bump version by one of the gem
+    def self.update_gem_latest_version_by_one(gem_version)
+      current_version = Gem::Version.new gem_version
+      new_version = current_version.bump
+    rescue StandardError => error
+      puts "(WARNING) Unable to check latest gem version. #{error}".red
+      return new_version
+    end
+
+    # @summary
+    #   Update Gemfile with multigem
+    # @param [String] output_path
+    #   The location that the command is to be run from.
+    # @param [String] gem_name
+    #   The gem name
+    # @param [String] gemfury_token
+    #   The gemfury token
+    # @param [String] gemfury_user
+    #   The gemfury user
+    def self.update_gemfile_multigem(output_path, gem_name, gemfury_token, gemfury_user)
+      gem_file_name = 'Gemfile'
+      gem_source_line = "source \"https://#{gemfury_token}@gem.fury.io/#{gemfury_user}/\""
+      Dir.chdir(output_path) unless Dir.pwd == output_path
+
+      if gem_name.nil? == false && gemfury_token.nil? == false && gemfury_user.nil? == false # rubocop:disable Style/GuardClause
+        # Append the gem with new source location
+        gem_name = gem_name.chomp('"').reverse.chomp('"').reverse
+        File.open('/tmp/out.tmp', 'w') do |out_file|
```

Comment:
Should probably add error handling for file operations here or where this is used

Reply:
thank you.adding them